packages/flutter/test/cupertino/route_test.dart

```diff
@@ -960,8 +974,14 @@ void main() {
 
     // Finish the rest of the animation
     await tester.pump(const Duration(milliseconds: 350));
+    // Give time to the animation to finish and update its status to
+    // AnimationState.completed, so the reverse curved can be used in the next
+    // step.
+    await tester.pump(const Duration(milliseconds: 1));
```

Comment:
The issue was that the parent of the curved animation needs to change its status from `forward` to `completed` so it doesn't ignore when it the parents updates it to `reverse`

https://github.com/flutter/flutter/blob/1a905d508de1608ba68ee1bd4492e8cb3a428709/packages/flutter/lib/src/animation/animations.dart#L427-L431

This is how I managed to do it, I don't know if there is a better way to do that?

`tester.pump()` alone doesn't work. `tester.pumpAndSettle()` works but I believe it is because it has a default duration of 100ms.

Would you have a better approach?

Reply:
Sure I can do that. I don't mind the `100ms` but I felt it was a bit "hidden"